cpp/daal/src/algorithms/dtrees/forest/classification/oneapi/df_classification_train_hist_oneapi_impl.i

```diff
@@ -330,12 +338,12 @@ services::Status ClassificationTrainBatchKernelOneAPI<algorithmFPType, hist>::co
         args.set(1, treeOrder, AccessModeIds::read);
         args.set(2, nodeList, AccessModeIds::read);
         args.set(3, nodeIndices, AccessModeIds::read);
-        args.set(4, (int32_t)nodeIndicesOffset);
+        args.set(4, static_cast<int32_t>(nodeIndicesOffset));
```

Comment:
Are all those conversions to int32_t SDL compliant?

Reply:
added corresponding DAAL_ASSERTs, and also corresponding checks in compute before kernels call